File: python/run_cave_dataset.py
#!/usr/bin/env python
import copy
import numpy as np
import os
from os.path import expanduser
import pickle
from tqdm import tqdm

import gmm_d2d_registration_py
from utils.RToZYX import RToZYX
from utils.ZYXToR import ZYXToR
from utils.QuatToR import QuatToR
from utils.pose_compose import pose_compose
from utils.pose_inverse import pose_inverse
from utils.run_dataset import run_dataset
from utils.plot_results import plot_results



def main():

    FIRST_SCAN = 499
    LAST_SCAN = 890
    PREFIX = 'isoplanarhybrid_'
    DATASET = 'cave'
    NUM_COMPONENTS = 100

    transforms, ground_truths = run_dataset(
        DATASET, FIRST_SCAN, LAST_SCAN, PREFIX, NUM_COMPONENTS)

    plot_results(transforms, ground_truths, FIRST_SCAN, LAST_SCAN)


if __name__ == '__main__':
    main()

File: matlab/run_cave_dataset.m
DATASET = 'cave';
FIRST_SCAN = 500;
LAST_SCAN = 890;
NUM_COMPONENTS = 100;
PREFIX = ['isoplanarhybrid_'];

[transforms, ground_truths] = process_dataset(DATASET, FIRST_SCAN, LAST_SCAN, PREFIX, NUM_COMPONENTS);
plot_trajectories(transforms, ground_truths, FIRST_SCAN, LAST_SCAN);
